src/functions_math.jl

```diff
@@ -108,3 +113,131 @@ function symmetric_names(L::Tuple{Symbol,Symbol}, dims::Integer)
     end
     return compile_time_return_hack(names)
 end
+
+################################################
+# FFT
+
+for fun in (:fft, :ifft, :bfft)
+    plan_fun = Symbol(:plan_, fun)
+    @eval begin
+
+        function AbstractFFTs.$fun(A::NamedDimsArray{L}) where {L}
+            data = AbstractFFTs.$fun(parent(A))
+            return NamedDimsArray(data, wave_name(L))
+        end
+
+        function AbstractFFTs.$fun(A::NamedDimsArray{L,T,N}, dims) where {L,T,N}
+            numerical_dims = dim(A, dims)
+            data = AbstractFFTs.$fun(parent(A), numerical_dims)
+            newL = wave_name(L, numerical_dims)
+            return NamedDimsArray(data, newL)
+        end
+
+        """
+            $($fun)(A, :time => :freq, :x => :kx)
+
+        Acting on a `NamedDimsArray`, this specifies to take the transform along the dimensions
+        named `:time, :x`, and return an array with names `:freq` and `:kx` in their places.
+
+        If new names are not given, then the default is `:x => :x∿` and `:x∿ => :x`,
+        applied to all dimensions, or to those specified as usual, e.g. `$($fun)(A, (1,2))`
+        or `$($fun)(A, :time)`. The symbol "∿" can be typed by `\\sinewave<tab>`.
+        """
+        function AbstractFFTs.$fun(A::NamedDimsArray{L,T,N}, p::Pair{Symbol,Symbol}, ps::Pair{Symbol,Symbol}...) where {L,T,N}
+            numerical_dims = dim(A, (first(p), first.(ps)...))
+            data = AbstractFFTs.$fun(parent(A), numerical_dims)
+            newL = replace_names(L, p, ps...)
+            return NamedDimsArray(data, newL)
+        end
+
+        """
+            F = $($fun)(A, :name)
+            A∿ = F * A == $($fun)(A, :name)
+
+        FFT plans for `NamedDimsArray`s can select a dimension based on names,
+        but cannot specify the final name. The `A∿` returned here will have default name
+        `:name∿` in the transformed dimension.
+        """
+        function AbstractFFTs.$plan_fun(A::NamedDimsArray, dims = ntuple(d->d, ndims(A)); kw...)
+            numerical_dims = dim(A, dims)
+            return AbstractFFTs.$plan_fun(parent(A), numerical_dims; kw...)
+        end
+
+        AbstractFFTs.$plan_fun(A::NamedDimsArray, p::Pair{Symbol,Symbol}, ps::Pair{Symbol,Symbol}...) =
+            throw(ArgumentError(string($plan_fun, "(::NamedDimsArray, ::Pair...) is not implemented. ",
+                "FFT plans can only specify an input name, not an output.")))
```

Comment:
Per BlueStyle https://github.com/invenia/BlueStyle/blob/master/.mailmap#method-definitions

> Only use short-form function definitions when they fit on a single line

I suspect this code was written before we adopted the style-guide ?

Reply:
Or was doodled up for my own amusement. Easy to fix.